transformers/integration_tests/median_lift_test.go

```diff
@@ -0,0 +1,87 @@
+	"github.com/makerdao/vdb-mcd-transformers/transformers/events/median_lift"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared/constants"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/test_data"
+	"github.com/makerdao/vulcanizedb/libraries/shared/factories/event"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("MedianLift EventTransformer", func() {
+	medianLiftConfig := event.TransformerConfig{
+		TransformerName:   constants.MedianLiftTable,
+		ContractAddresses: test_data.MedianAddresses(),
+		ContractAbi:       constants.MedianABI(),
+		Topic:             constants.MedianLiftSignature(),
+	}
+
+	It("transforms Median Lift log events", func() {
+		blockNumber := int64(8936530)
+		medianLiftConfig.StartingBlockNumber = blockNumber
+		medianLiftConfig.EndingBlockNumber = blockNumber
+
+		test_config.CleanTestDB(db)
+
+		header, headerErr := persistHeader(db, blockNumber, blockChain)
+		Expect(headerErr).NotTo(HaveOccurred())
+
+		// TODO: fetch event from blockchain once one exists
+		// logFetcher := fetcher.NewLogFetcher(blockChain)
+		// logs, logErr := logFetcher.FetchLogs(
+		// 	event.HexStringsToAddresses(medianLiftConfig.ContractAddresses),
+		// 	[]common.Hash{common.HexToHash(medianLiftConfig.Topic)},
+		// 	header)
+		// Expect(logErr).NotTo(HaveOccurred())
+
+		eventLogs := test_data.CreateLogs(header.Id, []types.Log{test_data.RawMedianLiftLogWithFiveAccounts}, db)
+
+		transformer := event.ConfiguredTransformer{
+			Config:      medianLiftConfig,
+			Transformer: median_lift.Transformer{},
+		}.NewTransformer(db)
+
+		transformErr := transformer.Execute(eventLogs)
+		Expect(transformErr).NotTo(HaveOccurred())
+
+		var dbResults []MedianLiftModel
+		err := db.Select(&dbResults, `SELECT address_id, msg_sender, a_length FROM maker.median_lift ORDER BY id`)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(len(dbResults)).To(Equal(1))
+		dbResult := dbResults[0]
+
+		contractAddressID, contractAddressErr := shared.GetOrCreateAddress(test_data.EthMedianAddress(), db)
+		Expect(contractAddressErr).NotTo(HaveOccurred())
+		Expect(dbResult.AddressID).To(Equal(strconv.FormatInt(contractAddressID, 10)))
+
+		msgSenderAddressID, msgSenderAddressErr := shared.GetOrCreateAddress("0xc45E7858EEf1318337A803Ede8C5A9bE12E2B40f", db)
+		Expect(msgSenderAddressErr).NotTo(HaveOccurred())
+		Expect(dbResult.MsgSender).To(Equal(strconv.FormatInt(msgSenderAddressID, 10)))
+
+		Expect(dbResult.ALength).To(Equal("5"))
+
+		var addresses []string
+		addressesError := db.Get(pq.Array(&addresses), `SELECT a FROM maker.median_lift ORDER BY id`)
+		Expect(addressesError).NotTo(HaveOccurred())
+		Expect(addresses).To(ConsistOf(
+			"0x6bDbc0ccC17d72a33Bf72a4657781a37DC2aa94E",
+			"0x26c45f7B0E456E36fC85781488A3CD42A57CcbD2",
+			"0x20c576F989EE94E571F027b30314aCF709267F7C",
+			"0xFCb1fB52E114b364B3Aab63d8a6f65Fe8dcbeF9D",
```

Comment:
Apologies for being out of the loop here but have we discussed whether we might want this array to contain foreign keys rather than text? Wondering if we'll want to use these addresses for spinning up/dropping medianizer transformers and, if so, whether we might need the hashed address from that table

Reply:
Can totally handle this later though, especially if there are other merged batch transformers already following this pattern

Reply:
I could be mistaken, but I don't _think_ these addresses correspond to contracts we'll want to track. I initially thought it would be a better idea to store the actual addresses rather than a FK, since Postgres can't enforce referential integrity on arrays of FKs. So it just seemed like going down that road would be asking for trouble (out-of-date references etc). Definitely down to rethink that decision though.

Reply:
Cool, the referential integrity thing is valid in it's own right, and 💯 if we don't need to track the addresses for secondary reasons